### PR TITLE
Gh 177/transaction details type and status

### PR DESCRIPTION
**Pull request by MouazAlzahabi**

### CODE REVIEW

File: Multisig/UI/Onboarding/TermsView.swift [hunk 1]
```diff
@@ -60,7 +60,7 @@ struct TermsView: View {
 
         var body: some View {
             HStack(alignment: .top) {
-                Image("ico-bullet-point")
+                Image("ico-bullet-point").foregroundColor(.gnoHold)
```
Comment (sche) [inline Multisig/UI/Onboarding/TermsView.swift:63]: can you do it on a separate line to align with '.padding'?

File: Multisig/UI/Transactions/TransactionDetails/TransactionStatusTypeView.swift [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+            if displayStatusOnSecondLine {
+                TransactionStatusView(status: transaction.status)
+            }
+        }
+    }
+
+    var title: String {
+        if let _ = transaction as? CustomTransaction {
+            return "Custom transaction"
+        }
```
Comment (sche) [inline Multisig/UI/Transactions/TransactionDetails/TransactionStatusTypeView.swift:33]: please put `else` in the same line after '}'

File: Multisig/UI/Transactions/TransactionDetails/TransactionStatusTypeView.swift [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+        else if let transfer = transaction as? TransferTransaction {
+            return transfer.isOutgoing ? "ico-outgoing-tx" : "ico-incoming-tx"
+        }
+
+        return "ico-settings-tx"
+    }
+
+    var displayStatusOnSecondLine: Bool {
+        if [TransactionStatus.waitingConfirmation, TransactionStatus.waitingExecution].contains(transaction.status) {
```
Comment (sche) [inline Multisig/UI/Transactions/TransactionDetails/TransactionStatusTypeView.swift:53]: `if [TransactionStatus.waitingConfirmation, .waitingExecution]`

File: Multisig/UI/Transactions/TransactionStatusView.swift [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+//
+
+import SwiftUI
+
+struct TransactionStatusView: View {
+    let status: TransactionStatus
+    var style: Style = .body
+    var body: some View {
+        HStack (spacing: 6) {
+            if [TransactionStatus.waitingConfirmation, TransactionStatus.waitingExecution].contains(status) {
```
Comment (sche) [inline Multisig/UI/Transactions/TransactionStatusView.swift:16]: `[TransactionStatus.waitingConfirmation, .waitingExecution]`

File: Multisig/UI/Transactions/TransactionDetails/TransactionStatusTypeView.swift (hunk 3 above)
Comment (DmitryBespalov) [inline Multisig/UI/Transactions/TransactionDetails/TransactionStatusTypeView.swift]: could be just 
```
var foo: Bool {
  [TransactionStatus.waitingConfirmation, .waitingExecution].contains(transaction.status)
}
```

File: Multisig/UI/Transactions/TransactionStatusView.swift (hunk 4 above)
Comment (DmitryBespalov) [inline Multisig/UI/Transactions/TransactionStatusView.swift]: this seems to be repeating, time to extract it into a computed var in the "status" - to the "isPending" variable

Comment (DmitryBespalov) [reply Multisig/UI/Transactions/TransactionStatusView.swift]: or use "isQueued"

File: Multisig/UI/Transactions/TransactionStatusView.swift [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+                Image("ico-bullet-point").foregroundColor(statusColor)
+            }
+
+            if style == .body {
+                BodyText(status.title, textColor: statusColor)
+            }
+            else {
```
Comment (DmitryBespalov) [inline Multisig/UI/Transactions/TransactionStatusView.swift:23]: `} else {`

File: Multisig/UI/Transactions/TransactionStatusView.swift [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+                FootnoteText(status.title, color: statusColor)
+            }
+        }
+    }
+    
+    var statusColor: Color {
```
Comment (DmitryBespalov) [inline Multisig/UI/Transactions/TransactionStatusView.swift:29]: :+1:

File: Multisig/UI/Transactions/TransactionsList/TransactionCell/TransactionCellView.swift [hunk 7]
```diff
@@ -22,13 +22,13 @@ struct TransactionCellView: View {
                 FootnoteText(transaction.formattedDate, color: .gnoDarkGrey).opacity(opacity)
 
                 if [TransactionStatus.success, TransactionStatus.pending, TransactionStatus.canceled, TransactionStatus.failed].contains(transaction.status)  {
-                    FootnoteText(formatedStatusText, color: statusColor)
+                    TransactionStatusView(status: transaction.status, style: .footnote)
                 }
             }
 
             if [TransactionStatus.waitingConfirmation, TransactionStatus.waitingExecution].contains(transaction.status) {
```
Comment (DmitryBespalov) [inline Multisig/UI/Transactions/TransactionsList/TransactionCell/TransactionCellView.swift:29]: here, too, condition is repeated all over the place

File: Multisig/UI/Transactions/TransactionStatusView.swift (hunk 4 above)
Comment (DmitryBespalov) [reply Multisig/UI/Transactions/TransactionStatusView.swift]: or "isWaiting"

File: Multisig/UI/Transactions/TransactionDetails/TransactionStatusTypeView.swift [hunk 8]
```diff
@@ -41,16 +40,15 @@ struct TransactionStatusTypeView: View {
...
+        } else if let transfer = transaction as? TransferTransaction {
             return transfer.isOutgoing ? "ico-outgoing-tx" : "ico-incoming-tx"
         }
 
         return "ico-settings-tx"
     }
 
     var displayStatusOnSecondLine: Bool {
-        if [TransactionStatus.waitingConfirmation, TransactionStatus.waitingExecution].contains(transaction.status) {
+        if [.waitingConfirmation, .waitingExecution].contains(transaction.status) {
```
Comment (DmitryBespalov) [inline Multisig/UI/Transactions/TransactionDetails/TransactionStatusTypeView.swift:51]: the condition is repeated in multiple places, time to extract it in the computed variable of the TransactionStatus

**Pull request merged**